### PR TITLE
Fix: Collapsed state between redraws

### PR DESCRIPTION
**Pull request by BeyondEvil**

### CODE REVIEW

File: src/pytest_html/scripts/datamanager.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 
 class DataManager {
     setManager(data) {
         const collapsedCategories = [...getCollapsedCategory(data.renderCollapsed)]
+        const collapsedIds = []
         const tests = Object.values(data.tests).flat().map((test, index) => {
             const collapsed = collapsedCategories.includes(test.result.toLowerCase())
+            const id = `test_${index}`
+            if (collapsed) {
+                collapsedIds.push(id)
```
Comment (drRedflint) [inline src/pytest_html/scripts/datamanager.js:11]: I'd avoid updating the collapsedIds collection in the map.
Instead I would do it after the construction of collapsed collection:
`const collapsedIds = collapsed.map(({id}) => id)`

Comment (drRedflint) [reply src/pytest_html/scripts/datamanager.js:11]: or even better: 
setCollapsedIds(collapsed.map(({id}) => id))

Comment (BeyondEvil) [reply src/pytest_html/scripts/datamanager.js:11]: I actually had it that way at first. But thought it was better to not do two iterations over the same collection.

File: src/pytest_html/scripts/datamanager.js [hunk 2]
```diff
@@ -47,6 +53,10 @@ class DataManager {
     get environment() {
         return this.renderData.environment
     }
+
+    get initialSort() {
+        return this.data.initialSort
```
Comment (drRedflint) [inline src/pytest_html/scripts/datamanager.js:58]: `return [...this.data.initialSort]`

This prevents data.initialSort from becoming compromised by being updated by reference.

**Pull request merged**